specification/sbi_atee.adoc

```diff
@@ -101,35 +101,107 @@ This operation requires the host to convert a sufficient number of non-confident
+between TVM and the host. The NACL shared memory interface is between TSM and the host and TSM is responsible for writing any
+trap-related CSRs and GPRs needed by the host to handle the exception. TSM is also responsible for reading the returned result and
+forwarding it to the TVM. Further details about which CSRs and GPRS are used by the TSM and the host can be found in
+<<table_tsm_csr_updates_in_nacl>>. The layout of NACL shared memory is shown below as `struct nacl_shmem` and `scratch` space
+layout for TSM is shown as `struct tsm_shmem_scratch`.
 
-The layout of the shared-memory region can vary by TSM version. The host can determine the size and offset of the
-regions enumerated in `vcpu_register_set_id` by calling `sbi_tee_host_get_tvm_vcpu_num_register_sets()` to get the
-number of enumerated sets, and then `sbi_tee_host_get_tvm_vcpu_num_register_sets()` to determine the offset.
+[source, C]
+-------------------
+struct nacl_shmem {
+    /* Scratch space. The layout of this scratch space is defined by the particular function being
+     * invoked.
+     *
+     * For the `sbi_tee_host_run_tvm_vcpu()` function in the TEE-H extension, the layout of this 
+     * scratch space matches the `tsm_shmem_scratch` struct given below.
+     */
+    uint64_t scratch[256];
+    uint64_t _reserved[240];
+    /* Bitmap indicating which CSRs in `csrs` the host wishes to sync.
+     *
+     * Currently unused in the TEE-related extensions and will not be read or written by the TSM.
+     */
+    uint64_t dirty_bitmap[16];
+    /* Hypervisor and virtual-supervisor CSRs. The 12-bit CSR number is transformed into a 10-bit
+     * index by extracting bits `{csr[11:10], csr[8:0]}` since `csr[9:8]` is always 2'b10 for HS
+     * and VS CSRs.
+     *
+     * These CSRs may be updated by `sbi_tee_host_run_tvm_vcpu()` in the TEE-Host extension. See
+     * the documentation of `sbi_tee_host_run_tvm_vcpu()` for more details.
+     */
+    uint64_t csrs[1024];
+};
+
+struct tsm_shmem_scratch {
+    /* General purpose registers for a TVM guest.
+     *
+     * The TSM will always read or write the minimum number of registers in this set to complete
+     * the requested action. To avoid leaking information from the TVM, the TSM must follow the
+     * given rules.
+     *
+     * The TSM will write to these registers upon return from `sbi_tee_host_run_tvm_vcpu()` when:
+     *  - The vCPU takes a store guest page fault in an emulated MMIO region.
+     *  - The vCPU makes an ECALL that is to be forwarded to the host.
+     *
+     * The TSM will read from these registers when:
+     *  - The vCPU takes a load guest page fault in an emulated MMIO region.
+     */
+    uint64_t guest_gprs[32];
+    uint64_t _reserved[224];
+};
+-------------------
+
+Given is the list of CSRs and GPRs that the TSM and the host are supposed to use from NACL shared memory. Also given
```

Comment:
More formal language preferred. something like

The below table describes the list of..
It also describes the operation allowed for each entity